oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/util/ThrottlingDocumentStoreWrapper.java

```diff
@@ -0,0 +1,227 @@
+    }
+
+    @Override
+    public <T extends Document> List<T> createOrUpdate(final Collection<T> collection, final List<UpdateOp> updateOps) {
+        performThrottling(collection);
+        return store.createOrUpdate(collection, updateOps);
+    }
+
+    @Override
+    public <T extends Document> T findAndUpdate(final Collection<T> collection, final UpdateOp update) {
+        performThrottling(collection);
+        return store.findAndUpdate(collection, update);
+    }
+
+    @Override
+    public CacheInvalidationStats invalidateCache() {
+        return store.invalidateCache();
+    }
+    
+    @Override
+    public CacheInvalidationStats invalidateCache(Iterable<String> keys) {
+        return store.invalidateCache(keys);
+    }
+
+    @Override
+    public <T extends Document> void invalidateCache(Collection<T> collection, String key) {
+        store.invalidateCache(collection, key);
+    }
+
+    @Override
+    public void dispose() {
+        store.dispose();
+    }
+
+    @Override
+    public <T extends Document> T getIfCached(final Collection<T> collection, final String key) {
+        return store.getIfCached(collection, key);
+    }
+
+    @Override
+    public void setReadWriteMode(String readWriteMode) {
+        store.setReadWriteMode(readWriteMode);
+    }
+
+    @Override
+    public Iterable<CacheStats> getCacheStats() {
+        return store.getCacheStats();
+    }
+
+    @Override
+    public Map<String, String> getMetadata() {
+        return store.getMetadata();
+    }
+
+    @NotNull
+    @Override
+    public Map<String, String> getStats() {
+        return store.getStats();
+    }
+
+    @Override
+    public long determineServerTimeDifferenceMillis() {
+        return store.determineServerTimeDifferenceMillis();
+    }
+
+    /**
+     * Return the size limit for node name based on the document store implementation
+     *
+     * @return node name size limit
+     */
+    @Override
+    public int getNodeNameLimit() {
+        return store.getNodeNameLimit();
+    }
+
+    /**
+     * Return the {@link Throttler} for the underlying store
+     * Default is no throttling
+     *
+     * @return throttler for document store
+     */
+    @Override
+    public Throttler throttler() {
+        return store.throttler();
+    }
+
+    // helper methods
+
+    private <T extends Document> void performThrottling(final Collection<T> collection) {
+
+        if (CLUSTER_NODES == collection) {
+            return;
+        }
+
+        final Throttler throttler = throttler();
+        long throttleTime = throttler.throttlingTime();
+
+        if (throttleTime == 0) {
+            return; // no throttling
+        }
+
+        try {
+            // log message every 10 secs once to reduce noise
+            if (now().getSecond() - lastLogTime >= 10) {
+                LOG.warn("Throttling the system for {} ms for {} collection", throttleTime, collection);
+                lastLogTime = now().getSecond();
+            }
```

Comment:
now().getSecond() seems to only return `0..59` though, so might not work as expected?
What about using decaseconds? 

PS: the initialization could then be `lastLogTime = 0`

```suggestion
            // (decaseconds since 1970 - overflows roughly in year 2650)
            final int currentDecaseconds = (int) (System.currentTimeMillis() / 10000);
            if (currentDecaseconds - lastLogTime >= 1) {
                lastLogTime = currentDecaseconds;
                LOG.warn("Throttling the system for {} ms for {} collection", throttleTime, collection);
            }
```